pytensor/link/pytorch/linker.py

```diff
@@ -0,0 +1,85 @@
+from numpy.random import Generator, RandomState
+
+from pytensor.compile.sharedvalue import SharedVariable, shared
+from pytensor.link.basic import JITLinker
+
+
+class PytorchLinker(JITLinker):
+    """A `Linker` that compiles NumPy-based operations using torch.compile."""
+
+    def fgraph_convert(self, fgraph, input_storage, storage_map, **kwargs):
+        from pytensor.link.pytorch.dispatch import pytorch_funcify
+        from pytensor.tensor.random.type import RandomType
+
+        shared_rng_inputs = [
+            inp
+            for inp in fgraph.inputs
+            if (isinstance(inp, SharedVariable) and isinstance(inp.type, RandomType))
+        ]
+
+        # Replace any shared RNG inputs so that their values can be updated in place
```

Comment:
This was JAX specific. We'll probably need a different approach for RandomVariables in Pytorch. How do they look like in PyTorch? I suggest a separate PR for these